libvast/builtins/commands/rebuild.cpp

```diff
@@ -277,28 +277,19 @@ struct rebuilder_state {
       .then(
         [this, finish](system::catalog_lookup_result& lookup_result) mutable {
           for (auto& [type, result] : lookup_result.candidate_infos) {
-            if (!run->options.all) {
-              std::erase_if(result.partition_infos,
-                            [&](const partition_info& partition) {
-                              return partition.version
-                                     >= version::current_partition_version;
-                            });
-              if (result.partition_infos.empty()) {
-                lookup_result.candidate_infos.erase(type);
-                if (lookup_result.candidate_infos.empty()) {
-                  return finish({}, true);
-                }
-              }
-            }
-            if (run->options.undersized) {
-              std::erase_if(
-                result.partition_infos, [&](const partition_info& partition) {
-                  return static_cast<bool>(partition.schema)
-                         && partition.events > detail::narrow_cast<size_t>(
-                              detail::narrow_cast<double>(max_partition_size)
-                              * undersized_threshold);
-                });
-            }
+            std::erase_if(
+              result.partition_infos, [&](const partition_info& partition) {
+                const auto erase_if_undersized
+                  = run->options.undersized
+                    && static_cast<bool>(partition.schema)
+                    && partition.events > detail::narrow_cast<size_t>(
+                         detail::narrow_cast<double>(max_partition_size)
+                         * undersized_threshold);
```

Comment:
I believe we need to do the inverse: similar to "erase_if_**not**_outdated", we want to remove items from the list if they are **not** undersized.

Additionally, the predicate `static_cast<bool>(partition.schema)` is not clear to me, what does it achieve?

Reply:
Ah, the check is actually correct, but the variable name should still reflect that.

Reply:
> `static_cast<bool>(partition.schema)`

I think we can actually remove that now, as VAST no longer supports reading partitions that have a heterogeneous schema. This was the check that indicated whether a partition was homogeneous.